packages/ui/components/formbuilder/formBuilderField.test.tsx

```diff
@@ -0,0 +1,133 @@
+  return {
+    ...actual,
+    useFormContext: vi.fn().mockReturnValue({
+      control: {} as Control, // Mock Control object
+      formState: {} as any, // Mock formState object
+    }),
+  };
+});
+
+describe("FormBuilderField: Disable on Prefill Behavior", () => {
+  beforeEach(() => {
+    // Reset mocks before each test
+    vi.resetAllMocks();
+  });
+
+  for (const { questionType, label } of questionTypes) {
+    test(`should remain enabled when 'disableOnPrefill' is true and there are errors for ${label} (${questionType})`, () => {
+      const field = {
+        value: "",
+        disableOnPrefill: true,
+        type: questionType,
+        name: `test-${questionType}-n`,
+        label: `test-${questionType}-l`,
+        placeholder: `test-${questionType}-p`,
+        required: true,
+      } as RhfFormField;
+      const formState = getFormatStateForQuestionType(questionType);
+      const mockFormContext = { formState, control: {} as Control };
+      // Mock `useFormContext` for this specific test
+      (vi.mocked(useFormContext) as any).mockReturnValue(mockFormContext);
+
+      const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
+
+      expect(shouldBeDisabled).toBe(false);
+    });
+  }
+  for (const { questionType, label } of questionTypes) {
+    test(`should remain enabled when 'disableOnPrefill' is false for ${label} (${questionType})`, () => {
+      const field = {
+        value: "",
+        disableOnPrefill: false,
+        type: questionType,
+        name: `test-${questionType}-n`,
+        label: `test-${questionType}-l`,
+        placeholder: `test-${questionType}-p`,
+        required: true,
+      } as RhfFormField;
+
+      const formState = getFormatStateForQuestionType(questionType);
+      const mockFormContext = { formState, control: {} as Control };
+      // Mock `useFormContext` for this specific test
+      (vi.mocked(useFormContext) as any).mockReturnValue(mockFormContext);
+
+      const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
+      expect(shouldBeDisabled).toBe(false);
+    });
+  }
+  for (const { questionType, label } of questionTypes) {
+    test(`should be disabled when 'disableOnPrefill' is true and there are no errors for ${label} (${questionType})`, () => {
+      const field = {
+        value: "",
+        disableOnPrefill: true,
+        type: questionType,
+        name: `test-${questionType}-n`,
+        label: `test-${questionType}-l`,
+        placeholder: `test-${questionType}-p`,
+        required: true,
+      } as RhfFormField;
+      //no errors
+      const formState = {};
+      const mockFormContext = { formState, control: {} as Control };
+      // Mock `useFormContext` for this specific test
+      (vi.mocked(useFormContext) as any).mockReturnValue(mockFormContext);
+      const shouldBeDisabled = useShouldBeDisabledDueToPrefill(field);
+      expect(shouldBeDisabled).toBe(true);
+    });
+  }
+});
```

Comment:
We need a test for `searchParams` as well. Input is only disabled if there are matching query params